README.md

```diff
@@ -24,86 +24,153 @@
+           ST_Transform(s.geom, 32129),
+           ST_Transform(bg.geom, 32129),
+           800
+       )
+),
+
+septa_bus_stop_surrounding_population as (
+
+   select
+       stop_id,
+       sum(p001001) as estimated_pop_800m
+   from septa_bus_stop_block_groups as s
+   join census_population as p using (geo_id)
+   group by stop_id
+)
+
+select
+   stop_name,
+   estimated_pop_800m,
+   geom
+from septa_bus_stop_surrounding_population
+join septa_bus_stops using (stop_id)
+order by estimated_pop_800m desc
+limit 1;
 
-  **The queries to #1 & #2 should generate relations with a single row, with the following structure:**
+2. Which bus stop has the smallest population within 800 meters?
 
-  ```sql
-  (
-      stop_name text, -- The name of the station
-      estimated_pop_800m integer, -- The population within 800 meters
-      the_geom geometry(Point, 4326) -- The geometry of the bus stop
-  )
-  ```
+Charter & Norcom
+
+with septa_bus_stop_block_groups as (
+   select
+       s.stop_id,
+       '1500000US' || bg.geoid10 as geo_id
+   from septa_bus_stops as s
+   join census_block_groups_2010 as bg
+       on ST_DWithin(
+           ST_Transform(s.geom, 32129),
+           ST_Transform(bg.geom, 32129),
+           800
+       )
+),
+
+septa_bus_stop_surrounding_population as (
+
+   select
+       stop_id,
+       sum(p001001) as estimated_pop_800m
+   from septa_bus_stop_block_groups as s
+   join census_population as p using (geo_id)
+   group by stop_id
+)
+
+select
+   stop_name,
+   estimated_pop_800m,
+   geom
+from septa_bus_stop_surrounding_population
+join septa_bus_stops using (stop_id)
+order by estimated_pop_800m asc
+limit 1;
 
 3. Using the Philadelphia Water Department Stormwater Billing Parcels dataset, pair each parcel with its closest bus stop. The final result should give the parcel address, bus stop name, and distance apart in meters. Order by distance (largest on top).
 
-  **Structure:**
-  ```sql
-  (
-      address text,  -- The address of the parcel
-      stop_name text,  -- The name of the bus stop
-      distance_m double precision  -- The distance apart in meters
-  )
-  ```
+Logic:
+- This one I'm less clear on, but would be most inclied to start by making centroids of each parcel
+- Next, I would see if a nearest neighbor function exists in SQL to join the centroid to its nearest neighbor
+- I would want my query to select the parcel address, bus stop name, and distance apart in meters and be in descending order
 
 4. Using the _shapes.txt_ file from GTFS bus feed, find the **two** routes with the longest trips. In the final query, give the `trip_headsign` that corresponds to the `shape_id` of this route and the length of the trip.
 
-  **Structure:**
-  ```sql
-  (
-      trip_headsign text,  -- Headsign of the trip
-      trip_length double precision  -- Length of the trip in meters
-  )
-  ```
+ Logic:
+ - First would use ST_Makeline to make lines based off of the endpoints of the route
```

Comment:
Why just the endpoints? Say the route were a loop; in that case the start and end points would be the same (or very close), so it would register as a very short trip, regardless of the total length of the route. Something like `ST_MakeLine` can operate on multiple points (rather than just two endpoints) to construct a path or linestring. IMHO, the function is misnamed; it would be better as `ST_MakeLineString` or `ST_MakePath`.